src/openloco/windows/constructionwnd.cpp

```diff
@@ -31,7 +31,7 @@ namespace openloco::ui::windows
 
         registers regs;
         regs.edx = widgetIndex;
-        regs.esi = (int32_t)&w;
+        regs.esi = (intptr_t)&w;
```

Comment:
Without those cast, when the code looks as in the suggestion:
```suggestion
        regs.esi = &w;
```
there's a warning:
```
error: invalid conversion from ‘openloco::ui::window*’ to ‘intptr_t’ {aka ‘int’} [-fpermissive]
         regs.esi = &w;
                    ^~
```

Reply:
I can leave regs as `int32_t` and do something like 
```suggestion
        regs.esi = (int32_t)((intptr_t)&w);
```
but I don't think it's worth it in this form, this is just sweeping the warnings under the rug rather than dealing with them.

Reply:
But they don't make any sense in 64-bit, I would just ignore the warning.

Reply:
One thing to consider is that storing pointers in 32 bit types might cause problems if the allocator decides to return high memory.